cmake/ProjectCompilerPostConfig.cmake

```diff
@@ -33,9 +33,12 @@ macro(enable_errors errors)
 endmacro()
 
 
-IF (CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM" AND UNIX)
```

Comment:
Do you know if one could use this on Mac? If so, would it make sense to change the logic to `if WINDOWS ... else ....`

Reply:
I think there's a more modern way to do this with generator expressions.  I think the `if WINDOWS` approach is safer ([Mac should support the hyphen syntax](https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/fp-model-fp.html)).

Reply:
Great question, let me check that out.

Reply:
In Xyce we use this in our CMake, which works for Windows, Mac, and Linux:
```
if(WIN32)
  add_compile_options("$<IF:$<CXX_COMPILER_ID:IntelLLVM>,/fp:precise,>")
else()
  add_compile_options("$<IF:$<CXX_COMPILER_ID:IntelLLVM>,-fp-model=precise,>")
endif()
```

If you want to use a generator expression, it might work something like this:
```
add_compile_options("$<IF:$<CXX_COMPILER_ID:IntelLLVM>,$<IF:$<WIN32>,/fp:precise,-fp-model=precise,")
```
It's a bit more obscure and not fitting with the current CMake style in the file.  Which one do you prefer?

Reply:
`if .. else ..` seems easier to read, but no strong feelings here.